addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,718 @@
+                    'message': _('Missing MyDATA Tax Exemption Category for line %s.', line_no),
+                }
+            if len(line.tax_ids) == 1 and line.tax_ids.amount not in VALID_TAX_AMOUNTS:
+                errors[f'l10n_gr_edi_line_{line_no}_invalid_tax_amount'] = {
+                    'message': _('Invalid tax amount for line %s. The valid values are %s.',
+                                 line_no, ', '.join(str(tax) for tax in VALID_TAX_AMOUNTS)),
+                }
+        return errors
+
+    def _l10n_gr_edi_get_pre_error_string(self):
+        self.ensure_one()
+        pre_error = self._l10n_gr_edi_get_pre_error_dict()
+        error_messages = (error_val['message'] for error_val in pre_error.values())
+        return '\n'.join(error_messages)
+
+    @api.model
+    def _l10n_gr_edi_handle_send_result(self, result, xml_vals):
+        """
+        Handle the result object received from sending xml to myDATA.
+        Create the related error/sent document with the necessary values.
+        """
+        move_xml_map = self._l10n_gr_edi_create_move_xml_map(xml_vals)
+        move_ids = list(move_xml_map.keys())
+
+        if 'error' in result:
+            # If the request failed at this stage, it is probably caused by connection/credentials issues.
+            # In such case, we don't need to attach the xml here as it won't be helpful for the user.
+            for move in move_ids:
+                move._l10n_gr_edi_create_document_invoice_error(result['error'])
+            return
+
+        for result_id, result_dict in result.items():
+            move = move_ids[result_id]
+            xml_content = move_xml_map[move]
+            document_values = {**result_dict, 'xml_content': xml_content}
+            # Delete previous error documents
+            move.l10n_gr_edi_document_ids.filtered(lambda d: d.state in ('invoice_error', 'bill_error') and not d.attachment_id).unlink()
+            if 'error' in result_dict:
+                # In this stage, the sending process has succeeded, and any error we receive is generated from the MyDATA API.
+                # Previous error(s) without attachments (generated from pre-compute) are now useless and can be unlinked.
+                if move.is_sale_document(include_receipts=True):
+                    move._l10n_gr_edi_create_document_invoice_error(document_values)
+                else:
+                    move._l10n_gr_edi_create_document_bill_error(document_values)
+            else:
+                if move.is_sale_document(include_receipts=True):
+                    move._l10n_gr_edi_create_document_invoice_sent(document_values)
+                else:
+                    move._l10n_gr_edi_create_document_bill_sent(document_values)
+
+    def _l10n_gr_edi_send_invoices(self):
+        """ Send batch of invoices SendInvoice XML to MyDATA. """
+        xml_vals = self._l10n_gr_edi_get_invoices_xml_vals()
```

Comment:
Don't forget to lock the records before sending them (`_with_locked_records`) and commit right after the response + the update of the documents